### PR TITLE
Fix #27179

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix incorrect result for query with row-level security, prewhere and LowCardinality filter. Fixes #27179

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Need to exclude the test from parallel runs.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
`tests flaky check` - is expected because test is not concurrent with itself - it uses the same user

File: src/Storages/MergeTree/MergeTreeRangeReader.cpp [hunk 1]
```diff
@@ -859,22 +859,28 @@ static ColumnPtr combineFilters(ColumnPtr first, ColumnPtr second)
         return first;
     }
 
-    auto mut_first = IColumn::mutate(std::move(first));
-    FilterDescription firsrt_descr(*mut_first);
+    FilterDescription firsrt_descr(*first);
```
Comment (kitaisreal) [inline src/Storages/MergeTree/MergeTreeRangeReader.cpp:862]: @KochetovNicolai please change name `firsrt_descr` to something readable.

**Review by kitaisreal — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeRangeReader.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Storages/MergeTree/MergeTreeRangeReader.cpp:862]: ah, yes

**Review by kitaisreal — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
That's strange it was not backported. Let's do

**Pull request merged**